lib/view_component/base.rb

```diff
@@ -50,6 +50,17 @@ def config
 
     attr_accessor :__vc_original_view_context
 
+    # Compoents can have a cache key that is used to cache the rendered output.
+    #
+    # @return [String]
+    def cache_key
+      @vc_cache_key = if defined?(__vc_cache_args)
+        Digest::MD5.hexdigest(
+          __vc_cache_args.map { |method| send(method) }.join("-")
```

Comment:
How does this differ from Rails caching? Should we leverage their caching infrastructure to keep parity behavior wise?

Reply:
@BlakeWilliams solid idea so I looked at https://api.rubyonrails.org/v6.0.6.1/classes/AbstractController/Caching.html#method-i-view_cache_dependencies
and implemented similar logic, only is the class_attribute is leaking across classes, do you have an idea why and how would you fix that

Reply:
I have fixed the leaking